query/src/main/kotlin/jetbrains/exodus/query/metadata/OModelMetaData.kt

```diff
@@ -41,39 +39,17 @@ class OModelMetaData(
 
     override fun onAddAssociation(typeName: String, association: AssociationEndMetaData) {
         synchronized {
```

Comment:
Please do not use synchronized. It causes the carrier thread to get stuck in virtual thread implementation.

Reply:
Got it. Did not notice.